stdlib/public/core/RuntimeFunctionCounters.swift

```diff
@@ -0,0 +1,506 @@
+    UInt32(0), UInt32(0), UInt32(0), UInt32(0), UInt32(0),
+    UInt32(0), UInt32(0), UInt32(0), UInt32(0)
+  )
+
+  // Use counter name as index.
+  subscript(_ index: String) -> UInt32 {
+    get {
+      if let index = _RuntimeFunctionCounters.runtimtFunctionNameToIndex[index] {
+        return self[index]
+      }
+      fatalError("Unknown counter name: \(index)")
+    }
+
+    set {
+      if let index = _RuntimeFunctionCounters.runtimtFunctionNameToIndex[index] {
+        self[index] = newValue
+        return
+      }
+      fatalError("Unknown counter name: \(index)")
+    }
+  }
+
+  subscript(_ index: Int) -> UInt32 {
+    @inline(never)
+    get {
+      if (index >= _RuntimeFunctionCounters.numRuntimeFunctionCounters) {
+        fatalError("Counter index should be in the range " +
+          "0..<\(_RuntimeFunctionCounters.numRuntimeFunctionCounters)")
+      }
+      var tmpCounters = counters
+      var counter: UInt32 = withUnsafePointer(to: &tmpCounters) { ptr in
+        return ptr.withMemoryRebound(to: UInt32.self, capacity: 64) { buf in
+          return buf[index]
+        }
+      }
+      return counter
+    }
+
+    @inline(never)
+    set {
+      if (index >= _RuntimeFunctionCounters.numRuntimeFunctionCounters) {
+        fatalError("Counter index should be in the range " +
+          "0..<\(_RuntimeFunctionCounters.numRuntimeFunctionCounters)")
+      }
+      withUnsafeMutablePointer(to: &counters) {
+        $0.withMemoryRebound(to: UInt32.self, capacity: 64) {
+          $0[index] = newValue
+        }
+      }
+    }
+  }
+}
+
+extension _RuntimeFunctionCounters {
+  @_silgen_name("getObjectRuntimeFunctionCounters")
+  static internal func getObjectRuntimeFunctionCounters(
+    _ object: UnsafeRawPointer, _ result: inout _RuntimeFunctionCountersState)
+
+  @_silgen_name("getGlobalRuntimeFunctionCounters")
+  static internal func getGlobalRuntimeFunctionCounters(
+    _ result: inout _RuntimeFunctionCountersState)
+
+  @_silgen_name("setGlobalRuntimeFunctionCounters")
+  static internal func setGlobalRuntimeFunctionCounters(
+    _ state: inout _RuntimeFunctionCountersState)
+
+  @_silgen_name("setObjectRuntimeFunctionCounters")
+  static internal func setObjectRuntimeFunctionCounters(
+    _ object: UnsafeRawPointer,
+    _ state: inout _RuntimeFunctionCountersState)
+
+  @_silgen_name("setGlobalRuntimeFunctionCountersMode")
+  static
+  public // @testable
+  func setGlobalRuntimeFunctionCountersMode(enable: Bool) -> Bool
+
+  @_silgen_name("setPerObjectRuntimeFunctionCountersMode")
+  static
+  public // @testable
+  func setPerObjectRuntimeFunctionCountersMode(enable: Bool) -> Bool
+
+  /// Enable runtime function counters updates by the runtime.
+  static
+  public // @testable
+  func enableRuntimeFunctionCountersUpdates(
+    mode: (globalMode: Bool, perObjectMode: Bool) = (true, true)) {
+    _ = _RuntimeFunctionCounters.setGlobalRuntimeFunctionCountersMode(
```

Comment:
`_ =` should become unnecessary once the function is marked with a `@discardableResult` attribute.